js/core.js

```diff
@@ -4781,9 +4788,37 @@
                     inputDisplayMode = displayAnnot.selector_ux_mode;
                 }
 
+                /**
+                 * bulkForeignKeyCreate column is set based on the following rules:
+                 *   1. defined on display property in visible-columns
+                 *   2. foreign key annotation
+                 *   3. table-display annotation
+                 *   4. display annotation on table, schema, then catalog
+                 *   5. default heuristics used when computeBulkCreateForeignKeyObject is called
+                 */
+                // check display annotation on table/schema/catalog
+                // if true, don't set anything and let the heuristics be used
+                // if false, set to false to turn off heuristics
+                if (this.table._shouldUseBulkCreateForeignKey === false) bulkCreateConstraintName = false;
+
+                if (this.table.annotations.contains(module._annotations.TABLE_DISPLAY)) {
+                    var tableAnnotation = module._getRecursiveAnnotationValue(context, this.table.annotations.get(module._annotations.TABLE_DISPLAY).content);
+
+                    if (tableAnnotation.bulk_create_foreign_key_candidates !== null) {
```

Comment:
Same comment about allowing `null`. I think for all these checks it's better if you check for the exact types that we allow. For example here it could be

```tsx
var prop = tableAnnotation.bulk_create_foreign_key_candidates;

if (prop === null || prop === false || Array.isArray(prop)) {
  bulkCreateConstraintName = prop;
}
```